.NET/Microsoft.Recognizers.Text.DateTime/Parsers/CJK/BaseCJKDateParser.cs

```diff
@@ -0,0 +1,750 @@
+                                case Constants.TimexMonthFull:
+                                    date = referenceDate.AddMonths(-number);
+                                    break;
+                                case Constants.TimexYear:
+                                    date = referenceDate.AddYears(-number);
+                                    break;
+                                default:
+                                    return ret;
+                            }
+
+                            ret.Timex = $"{DateTimeFormatUtil.LuisDate(date)}";
+                            ret.FutureValue = ret.PastValue = date;
+                            ret.Success = true;
+                            return ret;
+                        }
+
+                        var afterMatch = this.config.AfterRegex.Match(suffix);
+                        if (afterMatch.Success && suffix.StartsWith(afterMatch.Value, StringComparison.Ordinal))
+                        {
+                            DateObject date;
+                            switch (unitStr)
+                            {
+                                case Constants.TimexDay:
+                                    date = referenceDate.AddDays(number);
+                                    break;
+                                case Constants.TimexWeek:
+                                    date = referenceDate.AddDays(7 * number);
+                                    break;
+                                case Constants.TimexMonthFull:
+                                    date = referenceDate.AddMonths(number);
+                                    break;
+                                case Constants.TimexYear:
+                                    date = referenceDate.AddYears(number);
+                                    break;
+                                default:
+                                    return ret;
+                            }
+
+                            ret.Timex = $"{DateTimeFormatUtil.LuisDate(date)}";
+                            ret.FutureValue = ret.PastValue = date;
+                            ret.Success = true;
+                            return ret;
+                        }
+                    }
+                }
+            }
+
+            return ret;
+        }
+
+        // Convert Chinese Number to Integer
+        private int ConvertChineseToNum(string numStr)
+        {
+            var num = -1;
+            var er = this.config.IntegerExtractor.Extract(numStr);
+            if (er.Count != 0)
+            {
+                if (er[0].Type.Equals(Number.Constants.SYS_NUM_INTEGER, StringComparison.Ordinal))
+                {
+                    num = Convert.ToInt32((double)(this.config.NumberParser.Parse(er[0]).Value ?? 0));
+                }
+            }
+
+            return num;
+        }
+
+        // convert Chinese Year to Integer
+        private int ConvertChineseYearToInteger(string yearChsStr)
```

Comment:
Minor: Such methods should be renamed to "CJK" instead of "Chinese".

Reply:
done